filebeat/tests/system/test_shutdown.py

```diff
@@ -22,18 +20,56 @@ def test_shutdown(self):
                 with open(nasa_log, 'w') as outfile:
                     for line in infile:
                         outfile.write(line)
-
-        self.render_config_template(
-                path=os.path.abspath(self.working_dir) + "/log/*",
-                ignore_older="1h"
-        )
-
         os.mkdir(self.working_dir + "/log/")
         self.copy_files(["logs/nasa-50k.log"],
                         source_dir="../files",
                         target_dir="log")
 
+    def test_shutdown(self):
+        """
+        Test starting and stopping Filebeat under load.
+        """
+
+        self.render_config_template(
+            path=os.path.abspath(self.working_dir) + "/log/*",
+            ignore_older="1h"
+        )
         for i in range(1,5):
             proc = self.start_beat(logging_args=["-e", "-v"])
             time.sleep(.5)
             proc.check_kill_and_wait()
+
+    def test_shutdown_wait_ok(self):
+        """
+        Test stopping filebeat under load and wait for publisher queue to be emptied.
+        """
+        self.render_config_template(
+            path=os.path.abspath(self.working_dir) + "/log/*",
+            ignore_older="1h",
+            shutdown_timeout="10s",
+        )
+        proc = self.start_beat(logging_args=["-e", "-v"])
+        time.sleep(.5)
+        proc.check_kill_and_wait()
+
+        log = self.get_log()
+        assert "Shutdown timeout 10s" in log
+        assert "Continue shutdown: All enqueued events being published." in log
```

Comment:
Could you also check the registry file that all events were presisted there (offset)?